lib/compat/wordpress-6.6/block-template-utils.php

```diff
@@ -111,3 +111,248 @@ function gutenberg_get_template_hierarchy( $slug, $is_custom = false, $template_
+	 *
+	 * @param WP_Block_Template[]|null $block_templates Return an array of block templates to short-circuit the default query,
+	 *                                                  or null to allow WP to run its normal queries.
+	 * @param array  $query {
+	 *     Arguments to retrieve templates. All arguments are optional.
+	 *
+	 *     @type string[] $slug__in  List of slugs to include.
+	 *     @type int      $wp_id     Post ID of customized template.
+	 *     @type string   $area      A 'wp_template_part_area' taxonomy value to filter by (for 'wp_template_part' template type only).
+	 *     @type string   $post_type Post type to get the templates for.
+	 * }
+	 * @param string $template_type Template type. Either 'wp_template' or 'wp_template_part'.
+	 */
+	$templates = apply_filters( 'pre_get_block_templates', null, $query, $template_type );
+	if ( ! is_null( $templates ) ) {
+		return $templates;
+	}
+
+	$post_type     = isset( $query['post_type'] ) ? $query['post_type'] : '';
+	$wp_query_args = array(
+		'post_status'         => array( 'auto-draft', 'draft', 'publish' ),
+		'post_type'           => $template_type,
+		'posts_per_page'      => -1,
+		'no_found_rows'       => true,
+		'lazy_load_term_meta' => false,
+		'tax_query'           => array(
+			array(
+				'taxonomy' => 'wp_theme',
+				'field'    => 'name',
+				'terms'    => get_stylesheet(),
+			),
+		),
+	);
+
+	if ( 'wp_template_part' === $template_type && isset( $query['area'] ) ) {
+		$wp_query_args['tax_query'][]           = array(
+			'taxonomy' => 'wp_template_part_area',
+			'field'    => 'name',
+			'terms'    => $query['area'],
+		);
+		$wp_query_args['tax_query']['relation'] = 'AND';
+	}
+
+	if ( ! empty( $query['slug__in'] ) ) {
+		$wp_query_args['post_name__in']  = $query['slug__in'];
+		$wp_query_args['posts_per_page'] = count( array_unique( $query['slug__in'] ) );
+	}
+
+	// This is only needed for the regular templates/template parts post type listing and editor.
+	if ( isset( $query['wp_id'] ) ) {
+		$wp_query_args['p'] = $query['wp_id'];
+	} else {
+		$wp_query_args['post_status'] = 'publish';
+	}
+
+	$template_query = new WP_Query( $wp_query_args );
+	$query_result   = array();
+	foreach ( $template_query->posts as $post ) {
+		$template = _build_block_template_result_from_post( $post );
+
+		if ( is_wp_error( $template ) ) {
+			continue;
+		}
+
+		if ( $post_type && ! $template->is_custom ) {
+			continue;
+		}
+
+		if (
+			$post_type &&
+			isset( $template->post_types ) &&
+			! in_array( $post_type, $template->post_types, true )
+		) {
+			continue;
+		}
+
+		$query_result[] = $template;
+	}
+
+	if ( ! isset( $query['wp_id'] ) ) {
+		/*
+		 * If the query has found some use templates, those have priority
+		 * over the theme-provided ones, so we skip querying and building them.
+		 */
+		$query['slug__not_in'] = wp_list_pluck( $query_result, 'slug' );
+		$template_files        = _gutenberg_get_block_templates_files( $template_type, $query );
```

Comment:
Same as above, I think we should mark what's being changed with a `@core-merge` comment or similar.

Reply:
There is no need here. I had to duplicate the whole `gutenberg_get_block_templates` just for this line override :/